src/main/java/jenkins/plugins/logstash/LogstashConsoleLogFilter.java

```diff
@@ -16,14 +16,16 @@
 public class LogstashConsoleLogFilter extends ConsoleLogFilter implements Serializable
 {
 
+  private hudson.EnvVars envVars;
   private static final Logger LOGGER = Logger.getLogger(LogstashConsoleLogFilter.class.getName());
 
   private transient Run<?, ?> run;
   public LogstashConsoleLogFilter() {}
 
-  public LogstashConsoleLogFilter(Run<?, ?> run)
+  public LogstashConsoleLogFilter(Run<?, ?> run, hudson.EnvVars envVars)
```

Comment:
I'm a bit confused. Who is going to initialize this value for the old-style builds? I only see a change for pipeline

Reply:
I planned to introduce this modif only for declarative pipeline as it is the STAGE_NAME that were missing in the environment variables previously.
For old-style builds, I guess that your are talking for instance about freestyle, the STAGE_NAME is not relevant. So I guess that the plugin should work as before.

Reply:
as @mwinter69 mentioned here https://github.com/jenkinsci/logstash-plugin/pull/92#discussion_r405707603 it will break the freestyle build use case